specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,212 @@
+
+    if (result == COREWEBVIEW2_HIT_TEST_RESULT_HTCAPTION)
+    {
+        return HTCAPTION;
+    }
+
+    break;
+    ....
+}
+....
+}
+```
+#### C#/WinRT
+```c#
+....
+private bool isCapturingMouse = false;
+....
+
+protected override void WndProc(ref Message m)
+{
+....
+
+if (m.Msg == WM_NCHITTEST)
+{
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    if (CompositionController.GetHitTestResultAtPoint(point) == 
+        CoreWebView2HitTestResult.Htcaption) 
+    {
+        m.Result = (IntPtr)HTCAPTION;
+    }
+}
+// The following code forwards mouse messages to the WebView2 - Start
+else if (m.msg >= WM_MOUSEFIRST && m.msg <= WM_MOUSELAST || m.msg == WM_NCRBUTTONDOWN) 
+{
+  
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    // IsPointInWebview is an example of a function which your app can implement that checks if 
+    // the mouse message point lies inside the rect of the visual that hosts the WebView
+    if (IsPointInWebView(point) || m.msg == WM_MOUSELEAVE || isCapturingMouse) 
+    {
+
+        if (m.msg == WM_NCRBUTTONDOWN && m.w_param == HTCAPTION) 
+        {
+            // capturing the mouse will allow us to begin forwarding messages 
+            // to webview
+            isCapturingMouse = true;
+            SetCapture(mainWindowHwnd);
+        }
+        else if (message ==  WM_RBUTTONUP && GetCapture() == mainWindowHwnd) 
+        {
+            // no longer need to capture the mouse for draggbale regions support 
+            // after this
+            isCapturingMouse = false;
+            ReleaseCapture();
+        }
+        
+        // forward mouse messages to WV
+        CompositionController.SendMouseInput(
+            (CoreWebView2MouseEventKind)message,
+            (CoreWebView2MouseEventVirtualKeys)GET_KEYSTATE_WPARAM(wParam),
+            mouseData,
+            point);
+
+        m.Result = 0
+        return;
+    }
+        
+}
+// The following code forwards mouse messages to the WebView2 - Start
+    ....
+}
+       
+```
+# API Details
+## Win32 C++
+```cpp
+/// This enum contains values representing possible regions a given
+/// point lies within
+typedef enum COREWEBVIEW2_HIT_TEST_RESULT {
+    /// For regions in the WV which have the CSS style 'app-region: drag' set
+    COREWEBVIEW2_HIT_TEST_RESULT_CAPTION = 2,
+    /// For regions in the WV which don't have the CSS style 'app-region: drag' set
+    COREWEBVIEW2_HIT_TEST_RESULT_CLIENT,
+    /// Out of bounds of the app window
+    COREWEBVIEW2_HIT_TEST_RESULT_NONE = 0
+} COREWEBVIEW2_HIT_TEST_RESULT;
+
+/// This interface includes the new API for enabling Webview support for hit-testing regions
+[uuid(42BF7BA5-917A-4C27-ADA1-EA6969854C16), object, pointer_default(unique)]
+interface ICoreWebView2CompositionController4 : ICoreWebView2CompositionController3 {
+    /// Takes in a point and returns a Hit-test result value as an out parameter
```

Comment:
This should also link to documentation about `app-region`. Looking on MDN it seems like its non-standard and so MDN doesn't have official docs for it already, is that right? MDN has a tooltip suggesting we update it with info about app-region. Or we could have WebView2 specific docs about `app-region` and link to those from here instead of MDN. If we have a separate ADO item tracking adding documentation for `app-region` that's good enough and we don't need additional changes for this.

Embedding the documentation about `app-region` in this reference documentation is another way to go although it might be weird since most end devs will be using `app-region` in their web content along with our WPF element or WinUI elements and not have to use this method.

Reply:
I added a section with a brief description and a link to some of our docs.